src/pages/index.js

```diff
@@ -19,64 +21,192 @@ import {
   addNewCardButton,
   settings,
 } from "../utils/constants.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+import Api from "../components/API.js";
+import { data } from "autoprefixer";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "50644799-43da-42c8-ae19-a67fa2a948b2",
+    "Content-Type": "application/json",
+  },
+});
 
 const previewImagePopup = new PopupWithImage({
   popupSelector: "#card-preview-modal",
 });
-const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
-  const cardData = {
-    name: data.title,
-    link: data.url,
-  };
-  const newCard = createCard(cardData);
-  cardList.addItem(newCard);
-  addCardFormElement.reset();
-  addFormValidator.disableButton();
-});
 
 const cardList = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (data) => {
       cardList.addItem(createCard(data));
     },
   },
   ".cards__list"
 );
 
+const confirmDeleteModal = new PopupWithConfirm(
+  "#delete-modal",
+  handleDeleteClick
+);
+confirmDeleteModal.setEventListeners();
+
+const editAvatarModal = new PopupWithForm("#edit-avatar-modal", (input) => {
+  editAvatarModal.setLoading("Saving...");
+  handleAvatarEditSubmit(input);
+});
+editAvatarModal.setEventListeners();
+
+const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
+  const cardData = {
+    name: data.title,
+    link: data.url,
+  };
+
+  addCardPopup.setLoading("Saving...");
+
+  api
+    .addCards(cardData)
+    .then((res) => {
+      const card = createCard(res);
+      cardList.addItem(card);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.close();
+      addCardPopup.setLoading("Save");
+    })
+    .catch((err) => {
+      console.error(`Error, could not add card: ${err}`);
+    })
+    .finally(() => {
+      addCardFormElement.reset();
```

Comment:
`addCardFormElement.reset();` is not needed here because you have it in line 75


You need to **reset** the form only  in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.